core/domain/exp_domain.py

```diff
@@ -364,19 +441,38 @@ class RuleSpec(object):
     """Value object representing a rule specification."""
 
     def to_dict(self):
+        """Returns a dict representing this RuleSpec domain object.
+
+        Returns:
+            dict. A dict, mapping all fields of RuleSpec instance.
+        """
         return {
             'rule_type': self.rule_type,
             'inputs': self.inputs,
         }
 
     @classmethod
     def from_dict(cls, rulespec_dict):
+        """Return a RuleSpec domain object from a dict.
+
+        Args:
+            rulespec_dict: dict. The dict representation of RuleSpec object.
+
+        Returns:
+            RuleSpec. The corresponding RuleSpec domain object.
+        """
         return cls(
             rulespec_dict['rule_type'],
             rulespec_dict['inputs']
         )
 
     def __init__(self, rule_type, inputs):
+        """Initializes a RuleSpec domain object.
+
+        Args:
+            rule_type: str. The type of rule.
+            inputs: dict. TODO
```

Comment:
Here, let's write:

     inputs: dict. The values of the parameters needed in order to fully specify the rule. The keys for this dict can be deduced from the relevant description field in extensions/interactions/rule_templates.json -- they are the ones enclosed in {{...}} braces.